src/Okta.Sdk/Internal/ExternalTokenProvider.cs

```diff
@@ -0,0 +1,53 @@
+﻿// <copyright file="ExternalTokenProvider.cs" company="Okta, Inc">
+// Copyright (c) 2014 - present Okta, Inc. All rights reserved.
+// Licensed under the Apache 2.0 license. See the LICENSE file in the project root for full license information.
+// </copyright>
+
+using System;
+using System.Threading.Tasks;
+
+namespace Okta.Sdk.Internal
+{
+    /// <summary>
+    /// External Token Provider.
+    /// </summary>
+    public class ExternalTokenProvider : IOAuthTokenProvider
```

Comment:
If devs will have to pass their own implementation of IOAuthTokenProvider, why do we need this?

Reply:
Because they are free to pass null instead of an instance of the implementation, but we expect it not to be null further